pkg/kubectl/sorting_printer.go

```diff
@@ -78,14 +99,17 @@ func (s *SortingPrinter) sortObj(obj runtime.Object) error {
 		return err
 	}
 	if len(values) == 0 {
-		return fmt.Errorf("couldn't find any field with path: %s", s.SortField)
+		return fmt.Errorf("couldn't find any field with path: %s", field)
+	}
+	f, err := massageJSONPath(field)
```

Comment:
Seems like this should be done before `parser.Parse(field)`  Is it not necessary for some reason?  Either way, can we move it up near the beginning of the method since it reads like input sanitizing?